src/libraries/System.Text.Json/ref/System.Text.Json.cs

```diff
@@ -978,7 +978,7 @@ public sealed partial class JsonSourceGenerationOptionsAttribute : System.Text.J
         public System.Text.Json.Serialization.JsonKnownNamingPolicy PropertyNamingPolicy { get { throw null; } set { } }
         public bool WriteIndented { get { throw null; } set { } }
     }
-    [System.Diagnostics.CodeAnalysis.RequiresDynamicCodeAttribute("JSON serialization and deserialization might require types that cannot be statically analyzed and might need runtime code generation. Use System.Text.Json source generation for native AOT applications.")]
+    [System.Diagnostics.CodeAnalysis.RequiresDynamicCodeAttribute("Types deriving from JsonConverterFactory cannot be statically analyzed and might require runtime code generation.")]
```

Comment:
```suggestion
    [System.Diagnostics.CodeAnalysis.RequiresDynamicCodeAttribute("JsonStringEnumConverter cannot be statically analyzed and might require runtime code generation. There is no built-in solution for serializing and deserializing enums as strings that is compatible with native AOT applications currently. See https://github.com/dotnet/runtime/issues/73124.")]
```
We should explicitly say that there is no good solution for this today so that people do not waste time trying to find it.

Reply:
I've updated the message to also include recommendation for a workaround (i.e. write a non-factory custom converter for the enum).